src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -1142,95 +789,62 @@ winrt::hstring CascadiaSettings::ApplicationDisplayName()
 
 winrt::hstring CascadiaSettings::ApplicationVersion()
 {
+    static constexpr auto formatVersion = [](const unsigned int a, const unsigned int b, const unsigned int c, const unsigned int d) {
+        return winrt::hstring{ wil::str_printf<std::wstring>(L"%u.%u.%u.%u", a, b, c, d) };
+    };
+
     try
     {
-        const auto package{ winrt::Windows::ApplicationModel::Package::Current() };
-        const auto version{ package.Id().Version() };
-        winrt::hstring formatted{ wil::str_printf<std::wstring>(L"%u.%u.%u.%u", version.Major, version.Minor, version.Build, version.Revision) };
-        return formatted;
+        const auto package = winrt::Windows::ApplicationModel::Package::Current();
+        const auto version = package.Id().Version();
+        return formatVersion(version.Major, version.Minor, version.Build, version.Revision);
     }
     CATCH_LOG();
 
-    // Try to get the version the old-fashioned way
+    // Fallback if Windows Terminal is run unpackaged.
+    // It queries the version information embedded in its own DLL.
     try
     {
-        struct LocalizationInfo
-        {
-            WORD language, codepage;
-        };
-        // Use the current module instance handle for TerminalApp.dll, nullptr for WindowsTerminal.exe
-        auto filename{ wil::GetModuleFileNameW<std::wstring>(wil::GetModuleInstanceHandle()) };
-        auto size{ GetFileVersionInfoSizeExW(0, filename.c_str(), nullptr) };
+        const auto filename = wil::GetModuleFileNameW(wil::GetModuleInstanceHandle());

```

Comment:
I tested this code to death. It's extracted from some code I've written ages ago and it's very robust afaik. (I tested it! 😄)

Reply:
This actually regresses version support 😉 

The reason we load the localizable version resource is to get this:

![image](https://user-images.githubusercontent.com/189190/133338618-f90b2475-83a0-4990-8f52-c4a8a0c37582.png)

instead of this:

![image](https://user-images.githubusercontent.com/189190/133338559-b718c010-bffa-486f-aee6-4f8747ebf522.png)


This is important when you're running off a branch build, because the localizable name includes:

![image](https://user-images.githubusercontent.com/189190/133338758-5c5f8f94-7a97-484d-af3d-a8f0627fdb09.png)


the branch

Reply:
(Was there an issue in the original code to warrant its replacement?)

Reply:
> (Was there an issue in the original code to warrant its replacement?)

I've seen the code we use in other projects before and it's often a mistake, because usually want the 4-digit version number for later parsing/comparisons. I failed to see the comment explaining that this was done deliberately.